app/src/main/java/com/github/braillesystems/learnbraille/ui/screens/browser/MarkerViewFragment.kt

```diff
@@ -18,9 +23,65 @@ import com.github.braillesystems.learnbraille.ui.views.BrailleDotsViewMode
 import com.github.braillesystems.learnbraille.ui.views.display
 import com.github.braillesystems.learnbraille.ui.views.dotsState
 import com.github.braillesystems.learnbraille.utils.*
+import com.karlotoy.perfectune.instance.PerfectTune
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+enum class NoteDuration(
+    val titleStrId: Int,
+    val dot3: Boolean,
+    val dot6: Boolean,
+    val valueMillis: Long
+) {
+    EIGHTH(titleStrId = R.string.note_duration_8th, dot3 = false, dot6 = false, valueMillis = 125),
+    QUARTER(titleStrId = R.string.note_duration_4th, dot3 = false, dot6 = true, valueMillis = 250),
+    HALF(titleStrId = R.string.note_duration_half, dot3 = true, dot6 = false, valueMillis = 500),
+    FULL(titleStrId = R.string.note_duration_full, dot3 = true, dot6 = true, valueMillis = 1000);
+
+    // Converts a note of any duration in Braille to the note of the current duration
+    fun modifiedNote(note: BrailleDots): BrailleDots = BrailleDots(
+        note.b1, note.b2, BrailleDot.valueOf(dot3),
+        note.b4, note.b5, BrailleDot.valueOf(dot6)
+    )
+}
+
+private val noteToFreq = mapOf(
+    MarkerType.NoteC to 261.63, // C4 https://pages.mtu.edu/~suits/notefreqs.html
+    MarkerType.NoteD to 293.66,
+    MarkerType.NoteE to 329.63,
+    MarkerType.NoteF to 349.23,
+    MarkerType.NoteG to 392.00,
+    MarkerType.NoteA to 440.00,
+    MarkerType.NoteB to 493.88
+)
 
 class MarkerViewFragment : AbstractFragmentWithHelp(R.string.browser_marker_view_help) {
 
+    private fun chooseNoteDurationDialog(
+        brailleDots: BrailleDots,
+        block: (NoteDuration) -> Unit
+    ) {
+        val input = RadioGroup(context)
+        var noteDuration = NoteDuration.EIGHTH
+        NoteDuration.values().map { duration ->
+            input.addView(RadioButton(context).apply {
+                text = getString(R.string.browser_represent_template).format(
+                    getString(duration.titleStrId),
+                    duration.modifiedNote(brailleDots).spelling
+                )
+                setOnClickListener { noteDuration = duration }
+            })
+        }
+
+        AlertDialog.Builder(context)
```

Comment:
Is `Ok` button unnecessary here? It looks like it is redundant and the only choice is enough

Reply:
You're right. I've removed the button and made some other small improvements. Now it looks like this

<img width=200 alt="screenshot" src="https://user-images.githubusercontent.com/23435506/131811469-0ef8e476-015e-4545-8dc3-6997498431db.png">